examples/lighting-app/nrfconnect/main/AppTask.cpp

```diff
@@ -426,13 +389,25 @@ void AppTask::StartBLEAdvertisementHandler(AppEvent * aEvent)
     }
 }
 
-#ifdef CONFIG_CHIP_NFC_COMMISSIONING
 void AppTask::ChipEventHandler(const ChipDeviceEvent * event, intptr_t /* arg */)
 {
     if (event->Type != DeviceEventType::kCHIPoBLEAdvertisingChange)
         return;
 
-    if (event->CHIPoBLEAdvertisingChange.Result == kActivity_Started)
+    if (event->CHIPoBLEAdvertisingChange.Result == kActivity_Stopped)
+    {
+#ifdef CONFIG_CHIP_NFC_COMMISSIONING
+        NFCMgr().StopTagEmulation();
+#endif
+#ifdef CONFIG_MCUMGR_SMP_BT
+        // After CHIPoBLE advertising stop, start advertising SMP in case Thread is enabled or there are no active CHIPoBLE
+        // connections (exclude the case when CHIPoBLE advertising is stopped on the connection time)
+        if (GetDFUOverSMP().IsEnabled() && (ConnectivityMgr().IsThreadProvisioned() || ConnectivityMgr().NumBLEConnections() == 0))
```

Comment:
```suggestion
        if (GetDFUOverSMP().IsEnabled() && !ConnectivityMgr().IsBLEAdvertisingEnabled())
```
Wouldn't it be equivalent based on what we have found (that this flag is cleared when the advertising is stopped due to a BLE connection).

Reply:
Hmm I think not. We were looking on the moment when we should unregister GATT service (so when we disable CHIPoBLE advertising and connection is not active). If it comes to the advertising we want to disable it on the connection, so applying your suggested change will result in starting SMP advertising on every CHIPoBLE advertising stop, so also when BLE connection is active, what doesn't seem to be expected, right? I would assume that we don't want to have any advertising enabled when connection is set.

Reply:
My point is that `ConnectivityMgr().IsBLEAdvertisingEnabled()` seemed to stay active even after establishing a connection to the CHIPoBLE service, but maybe you're right that checking the number of connections is safer.